### PR TITLE
Tool 1126 swiftpm cache

### PR DESCRIPTION
**Pull request by lpusok**

### CODE REVIEW

File: xcodecache/derived_data_path_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+func Test_xcodeProjectHash(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		want    string
+		wantErr bool
+	}{
+		{
+			name:    "normal xcodeproj path",
+			path:    "/Users/lpusok/Develop/samples/sample-apps-ios-swiftpm/sample-swiftpm.xcodeproj",
```
Comment (godrei) [inline xcodecache/derived_data_path_test.go]: pls replace local reference: `/Users/lpusok` -> `/Users/bitrise` / `$HOME`

File: xcodecache/swiftpm_cache.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package cache
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/bitrise-io/go-steputils/cache"
+)
+
+// CollectPackagesCache marks the Swift Package Manager packages to be added the cache
```
Comment (godrei) [inline xcodecache/swiftpm_cache.go]: Comment should be a full sentence: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

File: xcodecache/swiftpm_cache.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+// The directory cached is: $HOME/Library/Developer/Xcode/DerivedData/[PER_PROJECT_DERIVED_DATA]/SourcePackages
+func CollectPackagesCache(projectPath string) error {
+	projectDerivedData, err := xcodeProjectDerivedDataPath(projectPath)
+	if err != nil {
+		return fmt.Errorf("%s", err)
```
Comment (godrei) [inline xcodecache/swiftpm_cache.go]: return err

File: xcodecache/swiftpm_cache.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	}
+
+	projectSwiftPMDir := path.Join(projectDerivedData, "SourcePackages")
+
+	cache := cache.New()
+	cache.IncludePath(projectSwiftPMDir)
```
Comment (godrei) [inline xcodecache/swiftpm_cache.go]: is not there any marker file, which could invalidate the cache in case of change?

Comment (lpusok) [reply xcodecache/swiftpm_cache.go]: There is a dependencies-state.json that could be used, I did not fully check that it always changes if the contents/state change tough.

File: xcodecache/derived_data_path_test.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+func Test_xcodeProjectDerivedDataPath(t *testing.T) {
+	tests := []struct {
+		name        string
+		projectPath string
+		want        string
+		wantErr     bool
+	}{
+		{
+			name:        "normal xcodeproj path",
+			projectPath: "/Users/lpusok/Develop/samples/sample-apps-ios-swiftpm/sample-swiftpm.xcodeproj",
```
Comment (godrei) [inline xcodecache/derived_data_path_test.go]: pls replace your username with bitrise

File: xcodecache/swiftpm_cache.go (hunk 2 above)
Comment (godrei) [reply xcodecache/swiftpm_cache.go]: Comment should end with `.` char: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

**Pull request merged**